gitlab/http_engines/requests_engine.py

```diff
@@ -0,0 +1,15 @@
+from typing import Any
+
+import requests
+
+
+class RequestsEngine:
+
+    _client: requests.Session
+
+    def __init__(self, **kwargs: Any) -> None:
+        self._client = kwargs.pop("session", requests.Session())
```

Comment:
I'm thinking that on the specific implementation, it can be a proper keyword argument and not popped from `kwargs`, right? Or am I missing something?

Reply:
Makes totally sense. Pushed a change.